sklearn/naive_bayes.py

```diff
@@ -117,6 +118,303 @@ def predict_proba(self, X):
+    1.0
+    """
+    # TODO consider jll for each estimator
+    # TODO unify variable names with similar meaning 
+    #      ("distribution", "model", "nb"), ("column", "feature")
+
+    def __init__(self, distributions):
+        self.distributions = distributions
+        self.classes_ = None
+        self.n_features_ = None
+        self._fits = None
+        self._columns = None
+        self._df_columns = None
+        self._is_fitted = False
+
+    def fit(self, X, y):
+        """Fit Gaussian Naive Bayes according to X, y
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples
+            and n_features is the number of features.
+        y : array-like, shape (n_samples,)
+            Target values.
+
+        Returns
+        -------
+        self : object
+        """
+        self._validate_distributions(X)
+        self._check_X_y(X, y)
+
+        self.classes_ = np.unique(y)
+
+        self._fits = [
+            (nb.fit(_safe_indexing(X, features, axis=1), y), features)
+            for (_, nb, _), features
+            in zip(self.distributions, self._columns)]
+
+        self._is_fitted = True
+
+        return self
+
+    def _joint_log_likelihood(self, X):
+        """Calculate the posterior log probability of the samples X
+
+        Parameters
+        ----------
+        X : ndarray
+        
+        Returns
+        -------
+        jll : ndarray, shape (1, n_classes)
+
+        Raises
+        ------
+        NotFittedError
+            If estimators have not been fitted
+        """
+        if not self._is_fitted:
+            raise NotFittedError("Call the fit() method first "
+                                 "before calling predict().")
+
+        # Obtain the log priors from each fitted estimator
+        log_priors = [
+            nb.class_log_prior_
+            if hasattr(nb, 'class_log_prior_') else np.log(nb.class_prior_)
+            for nb, _ in self._fits]
+
+        # Take any class log prior from the estimators
+        if np.allclose(*log_priors):  
+            log_prior = log_priors[0]
+        else:
+            raise ValueError("Class priors for every estimator "
+                             "must be the same.")
+
+        # Obtain the jll of each fitted estimator
+        jlls = [nb._joint_log_likelihood(
+                    np.array(_safe_indexing(X, features, axis=1)))
+                for (nb, features) in self._fits]
+
+        # Stack these jlls to give us
+        # the shape (distribution, sample, class)
+        jlls = np.hstack([jlls])
+
+        # Subtract the class log prior from all the jlls
+        # but add it back after the summation
+        jlls = jlls - log_prior
+        jll = jlls.sum(axis=0) + log_prior
+
+        return jll
+
+    @property
+    def distributions_(self):
+        return [(name, distr) for name, distr, _ in self.distributions]
```

Comment:
In ColumnTransformer we have triples `(name, fitted estimator, columns)`. Is there a benefit to leaving the columns out here? Otherwise consistency is preferable.

Reply:
What I really wanted to do was ColumnTransformer's `_transformer` but it didn't turn out right 😅. 

```python
@property
def _transformers(self):
    """
    Internal list of transformer only containing the name and
    transformers, dropping the columns. This is for the implementation
    of get_params via BaseComposition._get_params which expects lists
    of tuples of len 2.
    """
    return [(name, trans) for name, trans, _ in self.transformers]
```

Let's continue this conversation in the above [comment](https://github.com/scikit-learn/scikit-learn/pull/16281#discussion_r376772469).

Reply:
The private `_transformers` in ColumnTransformer is there to facilitate some of the behaviours inherited from BaseComposition. It's a bit ugly, but no one has made it prettier yet!